packages/build/src/steps/run_core_steps.js

```diff
@@ -0,0 +1,199 @@
+import { logBuildStart } from '../log/messages/core.js'
+import { reportStatuses } from '../status/report.js'
+
+import { addCoreSteps } from './get.js'
+import { runSteps } from './run_steps.js'
+
+/**
+ * Runs specific core steps for a build and returns whether it succeeded or not.
+ *
+ * @param  {string[]} [buildSteps] - a string array of build steps to run
+ * @param  {object} [flags] - build configuration CLI flags
+ * @param  {string} [flags.config] - Path to the configuration file
+ * @param  {string} [flags.cwd] - Current directory. Used to retrieve the configuration file
+ * @param  {string} [flags.repositoryRoot] - Git repository root directory. Used to retrieve the configuration file.
+ * @param  {string} [flags.apiHost] - Netlify API endpoint
+ * @param  {string} [flags.token] - Netlify API token for authentication
+ * @param  {string} [flags.siteId] - Netlify Site ID
+ * @param  {string} [flags.deployId] - Netlify Deploy ID
+ * @param  {string} [flags.context] - Build context
+ * @param  {string} [flags.branch] - Repository branch
+ * @param  {boolean} [flags.dry=false] - Run in dry mode, i.e. printing steps without executing them
+ * @param  {string} [flags.nodePath] - Path to the Node.js binary to use in the build command and plugins
+ * @param  {boolean} [flags.buffer=false] - Buffer output instead of printing it
+ *
+ * @returns {object} buildResult
+ * @returns {boolean} buildResult.success - Whether build succeeded or failed
+ * @returns {number} buildResult.severityCode - Build success/failure status among:
+ * 0 (success), 1 (build cancelled), 2 (user error), 3 (plugin error), 4 (system error). Can be used as exit code.
+ * @returns {string[]} buildResult.logs - When using the `buffer` option, all log messages
+ */
+export const runCoreSteps = async (buildSteps, flags = {}) => {
+  const { errorMonitor, mode, logs, debug, ...flagsA } = startBuild(flags)
+  const errorParams = { errorMonitor, mode, logs, debug }
+
+  try {
+    const { netlifyConfig: netlifyConfigA, configMutations } = await executeBuildStep({
+      ...flagsA,
+      errorMonitor,
+      mode,
+      logs,
+      debug,
+      errorParams,
+      buildSteps,
+    })
+    const { success, severityCode } = getSeverity('success')
+
+    return { success, severityCode, netlifyConfig: netlifyConfigA, configMutations, logs }
+  } catch (error) {
+    const { severity } = await handleBuildError(error, errorParams)
+    const { success, severityCode } = getSeverity(severity)
+
+    return { success, severityCode, logs }
+  }
+}
+
+const startBuild = function (flags) {
+  const logs = getBufferLogs(flags)
+
+  logBuildStart(logs)
+
+  const { bugsnagKey, ...flagsA } = normalizeFlags(flags, logs)
+  const errorMonitor = startErrorMonitor({ flags: flagsA, logs, bugsnagKey })
+
+  return { ...flagsA, errorMonitor, logs }
+}
+
+const getBuildSteps = function (buildSteps) {
+  const allSteps = addCoreSteps([]).filter(({ coreStepId }) => buildSteps.includes(coreStepId))
```

Comment:
I just realised that [the `getSteps` method](https://github.com/netlify/build/blob/52964224edfd5be00106d7098b7eb9a94c1de208/packages/build/src/steps/get.js#L8) could also be used here, with the exact same syntax, and it would mean that you wouldn't need to export `addCoreSteps`.

It would also be more consistent with the main flow.

(Sorry I didn't catch this in my previous review.)